app/controllers/posts_controller.rb

```diff
@@ -39,28 +41,57 @@ def create
     end
   end
 
+  # this will delete a post
+  def destroy
+    @post = Post.find(params[:id])
+    @user = @post.author
+    # delete all comments of this post
+    @post.comments.each(&:destroy)
+    # delete all likes of this post
+    @post.likes.each(&:destroy)
+    @post.destroy
+    respond_to do |format|
+      format.html do
+        flash[:notice] = 'Post was successfully deleted.'
+        redirect_to "/users/#{@user.id}"
+      end
+    end
+  end
+
   # this will create a comment
   def create_comment
     @comment = Comment.new(comment_params)
     @comment.post = Post.find(params[:id])
     @comment.author = current_user
 
     if @comment.save
-      flash[:success] = 'Comment created successfully'
+      flash[:notice] = 'Comment created successfully'
       redirect_to post_path
     else
-      flash.now[:error] = 'Error: Comment could not be created'
+      flash.now[:alert] = 'Error: Comment could not be created'
+    end
+  end
+
+  # this will delete a comment
+  def destroy_comment
+    @comment = Comment.find(params[:comment])
+    @comment.destroy
+    respond_to do |format|
+      format.html do
+        flash[:notice] = 'Comment was successfully deleted.'
+        redirect_to "/users/#{@comment.post.id}/posts/#{@comment.post.id}"
+      end
```

Comment:
- [ ] Comment management methods like `create` and `destroy` suppose to be in the `comment_Controller.rb` module not here. So please add this module with `new`, `create`, and `destroy` methods.